pymc/model_graph.py

```diff
@@ -39,6 +41,40 @@
 )
 
 
+@dataclass
+class DimInfo:
+    names: tuple[str | None]
+    sizes: tuple[int]
```

Comment:
size is usually the product of a shape or the dimension lengths that make a shape
```suggestion
    lengths: tuple[int]
```